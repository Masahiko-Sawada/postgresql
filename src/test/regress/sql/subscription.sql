--
-- SUBSCRIPTION
--

-- fail - no publications
CREATE SUBSCRIPTION testsub CONNECTION 'foo';

-- fail - no connection
CREATE SUBSCRIPTION testsub PUBLICATION foo;

CREATE SUBSCRIPTION testsub PUBLICATION testpub CONNECTION 'testconn' INITIALLY DISABLED NOCREATE_SLOT;

\dRs+

ALTER SUBSCRIPTION testsub PUBLICATION testpub2, testpub3;

\dRs

ALTER SUBSCRIPTION testsub PUBLICATION testpub, testpub1 CONNECTION 'conn2';

\dRs+

BEGIN;
ALTER SUBSCRIPTION testsub ENABLE;

\dRs

ALTER SUBSCRIPTION testsub DISABLE;

\dRs

COMMIT;

SET client_min_messages TO 'error';
DROP SUBSCRIPTION testsub;
RESET client_min_messages;
