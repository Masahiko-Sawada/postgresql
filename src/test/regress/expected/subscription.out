--
-- SUBSCRIPTION
--
-- fail - no publications
CREATE SUBSCRIPTION testsub CONNECTION 'foo';
ERROR:  publication not specified
-- fail - no connection
CREATE SUBSCRIPTION testsub PUBLICATION foo;
ERROR:  connection not specified
CREATE SUBSCRIPTION testsub PUBLICATION testpub CONNECTION 'testconn' INITIALLY DISABLED NOCREATE_SLOT;
\dRs+
           List of subscriptions
  Name   | Enabled | Publication | Conninfo 
---------+---------+-------------+----------
 testsub | f       | {testpub}   | testconn
(1 row)

ALTER SUBSCRIPTION testsub PUBLICATION testpub2, testpub3;
\dRs
          List of subscriptions
  Name   | Enabled |     Publication     
---------+---------+---------------------
 testsub | f       | {testpub2,testpub3}
(1 row)

ALTER SUBSCRIPTION testsub PUBLICATION testpub, testpub1 CONNECTION 'conn2';
\dRs+
               List of subscriptions
  Name   | Enabled |    Publication     | Conninfo 
---------+---------+--------------------+----------
 testsub | f       | {testpub,testpub1} | conn2
(1 row)

BEGIN;
ALTER SUBSCRIPTION testsub ENABLE;
\dRs
         List of subscriptions
  Name   | Enabled |    Publication     
---------+---------+--------------------
 testsub | t       | {testpub,testpub1}
(1 row)

ALTER SUBSCRIPTION testsub DISABLE;
\dRs
         List of subscriptions
  Name   | Enabled |    Publication     
---------+---------+--------------------
 testsub | f       | {testpub,testpub1}
(1 row)

COMMIT;
SET client_min_messages TO 'error';
DROP SUBSCRIPTION testsub;
RESET client_min_messages;
